dali/dfu/dfurun.cpp

```diff
@@ -1220,8 +1220,13 @@ class CDFUengine: public CInterface, implements IDFUengine
                             throw MakeStringException(-1,"Source file %s could not be found in Dali %s",tmp.str(),foreigndalinode->endpoint().getUrlStr(s).str());
                         }
                         kind.set(foreignfdesc->queryProperties().queryProp("@kind"));
-                        oldRoxiePrefix.set(foreignfdesc->queryProperties().queryProp("@roxiePrefix"));
                         iskey = strsame("key", kind);
+                        bool isForeignSuperfile = queryDistributedFileDirectory().isSuperFile(tmp.str(),foreignuserdesc,foreigndalinode);
```

Comment:
this approach works, but it's a bit costly..
It makes another call to the foreign Dali to get the file tree to check if it's a super, when it has already pulled all the info above (line 1217) when it called getFileDescriptor().
Instead of calling getFileDescriptor() why doesn't it call lookup() (in exactly same way that it would for non-local) to get a IDistributedFile ?
Then both could use the standard way to check for a super which is IDistributedFile::querySuperFile().

The foreign code to that calls getFileDescriptor() vs lookup() isn't new, but I can't see why it does it that way?